contracts/src/v0.8/dev/ArbitrumSequencerUptimeFeed.sol

```diff
@@ -0,0 +1,316 @@
+   */
+  constructor(address flagsAddress, address l1SenderAddress) {
+    setL1Sender(l1SenderAddress);
+
+    FLAGS = FlagsInterface(flagsAddress);
+  }
+
+  /**
+   * @notice Check if a roundId is valid in this current contract state
+   * @dev Mainly used for AggregatorV2V3Interface functions
+   * @param roundId Round ID to check
+   */
+  function isValidRound(uint256 roundId) private view returns (bool) {
+    return roundId > 0 && roundId <= type(uint80).max && s_feedState.latestRoundId >= roundId;
+  }
+
+  /// @notice Check that this contract is initialised, otherwise throw
+  function requireInitialized(uint80 latestRoundId) private pure {
+    if (latestRoundId == 0) {
+      revert Uninitialized();
+    }
+  }
+
+  /**
+   * @notice Initialise the first round. Can't be done in the constructor,
+   *    because this contract's address must be permissioned by the the Flags contract
+   *    (The Flags contract itself is a SimpleReadAccessController).
+   */
+  function initialize() external onlyOwner {
+    FeedState memory feedState = s_feedState;
+    if (feedState.latestRoundId != 0) {
+      revert AlreadyInitialized();
+    }
+
+    uint64 timestamp = uint64(block.timestamp);
+    bool currentStatus = FLAGS.getFlag(FLAG_L2_SEQ_OFFLINE);
+
+    // Initialise roundId == 1 as the first round
+    recordRound(1, currentStatus, timestamp);
+
+    emit Initialized();
+  }
+
+  /**
+   * @notice versions:
+   *
+   * - ArbitrumSequencerUptimeFeed 1.0.0: initial release
+   *
+   * @inheritdoc TypeAndVersionInterface
+   */
+  function typeAndVersion() external pure virtual override returns (string memory) {
+    return "ArbitrumSequencerUptimeFeed 1.0.0";
+  }
+
+  /// @return L1 sender address
+  function l1Sender() public view virtual returns (address) {
+    return s_l1Sender;
+  }
+
+  /**
+   * @notice Set the allowed L1 sender for this contract to a new L1 sender
+   * @dev Can be disabled by setting the L1 sender as `address(0)`. Accessible only by owner.
+   * @param to new L1 sender that will be allowed to call `updateStatus` on this contract
+   */
+  function transferL1Sender(address to) external virtual onlyOwner {
+    setL1Sender(to);
+  }
+
+  /// @notice internal method that stores the L1 sender
+  function setL1Sender(address to) private {
+    address from = s_l1Sender;
+    if (from != to) {
+      s_l1Sender = to;
+      emit L1SenderTransferred(from, to);
+    }
+  }
+
+  /**
+   * @dev Returns an AggregatorV2V3Interface compatible answer from status flag
+   *
+   * @param status The status flag to convert to an aggregator-compatible answer
+   */
+  function getStatusAnswer(bool status) private pure returns (int256) {
+    return status ? int256(1) : int256(0);
+  }
+
+  /**
+   * @notice The L2 xDomain `msg.sender`, generated from L1 sender address
+   */
+  function crossDomainMessenger() public view returns (address) {
```

Comment:
I agree, this name was confusing. I've changed it to `aliasedL1MessageSender` since it is Arbitrum's L2-aliased form of our stored L1 sender - does this make sense to you? I also updated the function natspec.